### ISSUE TITLE
Explain how to create a Flutter package with Rust (instead of an app)

### ISSUE THREAD

**Issue by fzyzcjy** [fzyzcjy/flutter_rust_bridge#845]
Currently we have a lot of doc if we want to create a flutter app with rust. but no doc or template yet for flutter packages.

related: https://github.com/fzyzcjy/flutter_rust_bridge/discussions/819#discussioncomment-4143007

**Comment by buraktabn** [comment]
Take a look at [Isar](https://github.com/isar/isar). I believe it can help what you're trying to achieve.

**Comment by fzyzcjy** [comment]
@buraktabn Hi thanks for the link! Indeed flutter_rust_bridge does support packages since day one - because that's how I personally use it. It is just lacking doc and template because my personal usage depends on my internal python-based tooling which is surely not suitable for external devs.

**Comment by GregoryConrad** [comment]
@fzyzcjy Since I have done this entire process already, using the latest recommended approaches where possible, I think I would be willing to take this issue on sometime soon. Isar is a good starting point (and indeed I used it for reference in certain places my library); however, I found better ways to take care of some things related to Rust interop that Isar does not do.

In the meantime, if anyone comes across this issue before I can tackle the documentation, take a look at my library, [mimir](https://github.com/gregoryconrad/mimir) for an example of how to make a Dart-only base and then a Flutter library built directly on top of it for ease of use.
